src/stats.jl

```diff
@@ -17,24 +17,43 @@ Setting `append_chains=false` will return a vector of dataframes containing the
 """
 function autocor(
     chains::Chains;
-    append_chains = true,
+    sections = _default_sections(chains),
+    append_chains::Bool = true,
     demean::Bool = true,
     lags::AbstractVector{<:Integer} = _default_lags(chains, append_chains),
-    kwargs...
+    var_names = nothing,
+    kwargs...,
 )
-    funs = Function[]
-    func_names = @. Symbol("lag ", lags)
-    for i in lags
-        push!(funs, x -> autocor(x, [i], demean=demean)[1])
-    end
+    chn = Chains(chains, _clean_sections(chains, sections))
 
-    return summarize(
-        chains, funs...;
-        func_names = func_names,
-        append_chains = append_chains,
-        name = "Autocorrelation",
-        kwargs...
-    )
+    # Obtain names of parameters.
+    names_of_params = var_names === nothing ? names(chn) : var_names
+
+    # set up the functions to be evaluated
```

Comment:
Maybe not use the phrase `functions` here? `lags` are just integers right? 
Maybe
```suggestion
    # Construct column names for lags.
```